pkg/kubectl/cmd/apiresources.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	unversioned_client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+// ApiResourcesOptions contains all the things necessary to run the api resources command.
+type ApiResourcesOptions struct {
+	groupVersion        string
+	discovery           unversioned_client.DiscoveryInterface
+	out                 io.Writer
+	noHeaders           bool
+	onlyNamespaced      bool
+	excludeSubresources bool
+}
+
+const (
+	apiResourcesLong = `Print the supported resources for an API version on the server.
+`
+
+	apiResourcesExample = `# Print the resources for API version 'v1'.
+$ kubectl api-resources v1
+
+# Print the resources for API version 'batch/v1' showing only namespaced resources,
+# excluding subresources.
+$ kubectl api-resources batch/v1 --only-namespaced --exclude-subresources`
+)
+
+// NewCmdApiResources creates a new api resources command.
+func NewCmdApiResources(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &ApiResourcesOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "api-resources GROUP_VERSION",
+		Short:   "Print the supported resources for an API version on the server.",
+		Long:    apiResourcesLong,
+		Example: apiResourcesExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args, out))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	cmd.Flags().BoolVar(&options.noHeaders, "no-headers", options.noHeaders, "When using the default output, don't print headers.")
+	cmd.Flags().BoolVar(&options.onlyNamespaced, "only-namespaced", options.onlyNamespaced, "Only show namespaced resources")
+	cmd.Flags().BoolVar(&options.excludeSubresources, "exclude-subresources", options.excludeSubresources, "Exclude subresources")
```

Comment:
maybe make this include-subresources and default to true so not expressed as a negative?